### ISSUE TITLE
[rush] HBO feedback from testing 5.60.0-rc.1 with phased commands

### ISSUE THREAD

**Issue by octogonz** [microsoft/rushstack#3172]
Notes

- [x] @elliot-nelson said that it wasn't immediately clear where to find this file.
   ```
   The config/rush-project.json file appears to be in an outdated format. 
   Please see the UPGRADING.md file in the Rush project for upgrade instructions.
   ```
   We should add a link in the README.md file like [Heft does](https://github.com/microsoft/rushstack/blob/master/apps/heft/README.md)

- [x] But also maybe we should make a handy redirect URL:  `Please see the UPGRADING.md notes for details. Quick link: https://rushjs.io/link/upgrading`

So far it seems to be working otherwise.

CC @iclanton

**Comment by elliot-nelson** [comment]
An observation (not necessarily something that needs fixing).

Prior to v5.59, it was still true that projects with disabled build caching would revert to the old _incremental_ build skipping logic.  That's no longer true.

In the past, I would set all of our projects that didn't have build scripts (things like config projects, rigs, etc.) to disable build cache, which meant that they always showed up in the "SKIPPED" category in a rush build.  Now, this doesn't work... if you do a `rush build` on a branch with no changes, you end up seeing all of the _real_ projects in the "FROM CACHE" section, and all the boilerplate projects in the "SUCCESSFUL" section.

To prevent that, the easy fix is to go into your `rush-project.json` files on those projects with no builds, and enable the build caching, and insert the section for defining the cache folders etc.  Even though the projects don't _produce_ any files, caching an empty file set does work now (I believe someone fixed that last year sometime).

This moves all the boilerplate projects into the "FROM CACHE" section, which is at least closer to the truth than the "SUCCESSFUL" section.

**Comment by elliot-nelson** [comment]
(If I had any comment for the future... I almost wonder if a phase with an empty build script should be immediately shunted into the "SKIPPED" category, and never even attempt to build or restore it, since it can't possibly produce artifacts.)

**Comment by iclanton** [comment]
@elliot-nelson - @dmichon-msft has a PR (https://github.com/microsoft/rushstack/pull/3043) that he thinks will address the issues that you've raised.

We also just briefly discussed creating a new status for projects with no scripts to disambiguate them from `FROM CACHE`/`SUCCESS`/`SKIPPED`

**Comment by dmichon-msft** [comment]
@elliot-nelson 

The PR addresses the ability to skip phases, modifies the treatment of empty scripts to resolve as "from cache" instead of "skipped" (with all the consequences that implies for subsequent projects), and ensures that task execution order always accounts for the full dependency relationship among selected tasks.

I don't know of any changes to the behavior of allowing a task to be skipped if the cache is disabled, that may require a separate investigation if we have a regression there.

**Comment by elliot-nelson** [comment]
@dmichon-msft I don't think it's a regression; my impression is that it was a conscious change to make sure that the "output" of a skipped incremental build could never be used as the input for a caching build.

To ensure this the best way is just to disable all incremental building if build caching is configured - I think that happened as part of the phasing implementation.

**Comment by octogonz** [comment]
Fixed in PR #3174
